packages/salesforcedx-vscode-core/src/commands/forceSourceDeployManifest.ts

```diff
@@ -34,6 +34,12 @@ export class ForceSourceDeployManifestExecutor extends BaseDeployExecutor {
 }
 
 export async function forceSourceDeployManifest(manifestUri: vscode.Uri) {
+  if (!manifestUri) {
+    const editor = vscode.window.activeTextEditor;
+    if (editor) {
```

Comment:
This will not always be `undefined` when triggering the command using keyboard shortcuts and the user doesn't have an open document. You can leverage the existing language id and check for it e.g.`if (editor && editor.document.languageId === 'forcesourcemanifest')`. We should also throw an informative error message when it's not a manifest file or the users will keep getting the `Cannot read property 'fsPath' of undefined` warning message.